src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/PaymentsCallbackPBAHandler.java

```diff
@@ -0,0 +1,149 @@
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.PaymentDetails;
+import uk.gov.hmcts.reform.civil.service.PaymentsService;
+import uk.gov.hmcts.reform.civil.service.Time;
+import uk.gov.hmcts.reform.payments.client.InvalidPaymentRequestException;
+import uk.gov.hmcts.reform.payments.client.models.PaymentDto;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.civil.callback.CallbackParams.Params.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.MAKE_PBA_PAYMENT;
+import static uk.gov.hmcts.reform.civil.enums.PaymentStatus.FAILED;
+import static uk.gov.hmcts.reform.civil.enums.PaymentStatus.SUCCESS;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PaymentsCallbackPBAHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(MAKE_PBA_PAYMENT);
+    private static final String ERROR_MESSAGE = "Technical error occurred";
+    private static final String TASK_ID = "CreateClaimMakePayment";
+    public static final String DUPLICATE_PAYMENT_MESSAGE
+        = "You attempted to retry the payment to soon. Try again later.";
+
+    private final PaymentsService paymentsService;
+    private final ObjectMapper objectMapper;
+    private final FeatureToggleService featureToggleService;
+    private final Time time;
+
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT), this::makePbaPayment
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        if (featureToggleService.isPbaV3Enabled()) {
+            return EVENTS;
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private CaseData updateWithDuplicatePaymentError(CaseData caseData) {
+        var paymentDetails = ofNullable(caseData.getClaimIssuedPaymentDetails())
+            .map(PaymentDetails::toBuilder)
+            .orElse(PaymentDetails.builder())
+            .status(FAILED)
+            .errorCode(null)
+            .errorMessage(DUPLICATE_PAYMENT_MESSAGE)
+            .build();
+
+        return caseData.toBuilder().claimIssuedPaymentDetails(paymentDetails).build();
+    }
+
+    private CallbackResponse makePbaPayment(CallbackParams callbackParams) {
+        var caseData = callbackParams.getCaseData();
+        var authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        List<String> errors = new ArrayList<>();
+        try {
+            log.info("processing payment for case " + caseData.getCcdCaseReference());
```

Comment:
Don't use concatenation in `log` statements. Use `{}` instead. See other comments.